common/thorhelper/thorread.cpp

```diff
@@ -39,6 +39,76 @@
     #include "parquetembed.hpp"
 #endif
 
+constexpr size32_t defaultReadBufferSize = 0x100000;
+
+//---------------------------------------------------------------------------------------------------------------------
+
+static IBufferedSerialInputStream * createInputStream(IFileIO * inputfileio, const IPropertyTree * providerOptions)
+{
+    assertex(providerOptions);
+    size32_t readBufferSize = providerOptions->getPropInt("readBufferSize", defaultReadBufferSize);
+
+    //MORE:
+    offset_t startOffset = 0;
+    offset_t length = (offset_t)-1;
+
+    //MORE: Is this a good idea?
+    if (length == unknownFileSize)
+    {
+        offset_t filesize = inputfileio->size();
+        assertex(startOffset <= filesize);
+        length = filesize - startOffset;
+    }
+
+    return createFileSerialStream(inputfileio, startOffset, length, readBufferSize);
+};
+
+
+static bool createInputStream(Shared<IBufferedSerialInputStream> & inputStream, Shared<IFileIO> & inputfileio, IFile * inputFile, const IPropertyTree * providerOptions)
+{
+    assertex(providerOptions);
+
+    bool compressed = providerOptions->getPropBool("@compressed", false);
+    bool blockcompressed = providerOptions->getPropBool("@blockCompressed", false);
+    bool forceCompressed = providerOptions->getPropBool("@forceCompressed", false);
+
+    MemoryBuffer encryptionKey;
+    if (providerOptions->hasProp("encryptionKey"))
+        providerOptions->getPropBin("encryptionKey", encryptionKey);
+
+    bool rowcompressed = false;
+    try
+    {
+        if (compressed)
+        {
+            Owned<IExpander> eexp;
+            if (encryptionKey.length()!=0)
+                eexp.setown(createAESExpander256((size32_t)encryptionKey.length(),encryptionKey.bufferBase()));
+            inputfileio.setown(createCompressedFileReader(inputFile,eexp));
```

Comment:
Style, trivial: spaces needed between args, here and elsewhere in this block.

Reply:
That's what comes from moving existing code...  I will fix.